modules/objdetect/src/qrcode.cpp

```diff
@@ -2370,4 +4701,1159 @@ bool QRCodeDetector::detectAndDecodeMulti(
+    vector<uint8_t>	payload;
+    /**rearranged data bits in encoding style*/
+    vector<uint8_t>	rearranged_data;
+
+    Mat original;
+    Mat masked_data;
+
+    uint32_t eci;
+    uint32_t fnc1_second_AI;
+    uint8_t parity;
+    uint8_t sequence_num;
+    uint8_t total_num;
+
+    bool fnc1_first;
+    bool fnc1_second;
+    /**basic information */
+    const VersionInfo *version_info ;
+    /**principles  about group and blocks*/
+    const  BlockParams *cur_ecc_params;
+    vector<Mat> my_qrcodes;
+public:
+    QREncoder(const std::string& input ,int mode,int v  ,int ecc  ,int mask  ,int eci_mode ,int structure_num );
+    Mat QRcodeGenerate();
+
+    void encodeByte(const std::string& input,vector<uint8_t>& output);
+    bool encodeAlpha(const std::string& input,vector<uint8_t>& output);
+    bool encodeNumeric(const std::string& input,vector<uint8_t>& output);
+    bool encodeKanji(const std::string& input,vector<uint8_t>& output);
+    bool encodeAuto(const std::string& input,vector<uint8_t>& output);
+    bool encodeECI(vector<uint8_t>& output);
+    bool encodeFNC1(const std::string& input,vector<uint8_t>& output);
+    bool encodeFNC2(const std::string& input,vector<uint8_t>& output);
+    bool encodeStructure(vector<uint8_t>& payload);
+
+    void padBitStream();
+    void stringToBits();
+    void eccGenerate(vector<Mat>& data_blocks,vector<Mat>& ecc_blocks);
+    void rearrangeBlocks(const vector<Mat>& data_blocks,const vector<Mat>& ecc_blocks);
+    void writeReservedArea();
+    void writeBit(int x, int y, int& count);
+    void writeData();
+    void structureFinalMessage();
+
+    void formatGenerate(const int& type , Mat & format_array);
+    void fillReserved(const Mat & format_array , Mat & masked);
+    void maskData(const int& type , Mat & masked);
+    void findAutoMaskType();
+
+    QRencodeMode fncModeSelect(const std::string& input);
+    bool versionEstimate(const int &input_length,vector<int>&possible_version);
+    bool generateBlock(const std::string& input, int mode ,methodBlock& block );
+    int versionAuto(const std::string & input_str);
+
+};
+/**findVersionCapacity
+ * @param input_length
+ * @param ecc_level
+ * @param mode
+ * @func  find the version automatically
+ * @return the version index
+ * */
+int findVersionCapacity(const int &input_length ,const int &ecc_level ,const int & version_begin, const int & version_end ){
+    int i,version_index,data_codewords;
+    version_index = -1;
+    /**find the version according to the number of data codewords*/
+    for( i = version_begin ; i < version_end ; i++ ){
+        const BlockParams * tmp_ecc_params = &(version_info_database[i].ecc[ecc_level]);
+        data_codewords = version_info_database[i].total_codewords
+                             -
+                             tmp_ecc_params->ecc_codewords*(tmp_ecc_params->num_blocks_in_G1+tmp_ecc_params->num_blocks_in_G2);
+
+        if(data_codewords*8 > input_length ){
```

Comment:
Replace all magic constants to variables with understandable name. It will make code more readable.